core/src/test/java/io/questdb/griffin/AlterTableSetTypeRestartTest.java

```diff
@@ -0,0 +1,384 @@
+                    "count\n" +
+                            count + "\n"
+            );
+        }
+    }
+
+    private static SqlExecutionContext createSqlExecutionContext(CairoEngine engine) {
+        return new SqlExecutionContextImpl(engine, 1).with(
+                AllowAllCairoSecurityContext.INSTANCE,
+                null,
+                null,
+                -1,
+                null
+        );
+    }
+
+    private static void createTable(String tableName, String walMode) throws SQLException {
+        runSqlViaPG("create table " + tableName + " (ts timestamp, x long) timestamp(ts) PARTITION BY DAY " + walMode);
+        LOG.info().$("created table: ").utf8(tableName).$();
+    }
+
+    private static Path doesConvertFileExist(CairoEngine engine, String tableName, boolean doesExist) {
+        final TableToken token = engine.getTableToken(tableName);
+        final Path path = Path.PATH.get().of(engine.getConfiguration().getRoot()).concat(token).concat(WalUtils.CONVERT_FILE_NAME);
+        MatcherAssert.assertThat(Chars.toString(path), Files.exists(path.$()), Matchers.is(doesExist));
+        return doesExist ? path : null;
+    }
+
+    private static void drainWalQueue(CairoEngine engine) {
+        try (final ApplyWal2TableJob walApplyJob = new ApplyWal2TableJob(engine, 1, 1, null)) {
+            walApplyJob.drain(0);
+            new CheckWalTransactionsJob(engine).run(0);
+            // run once again as there might be notifications to handle now
+            walApplyJob.drain(0);
+        }
+    }
+
+    private static void dropTable(String tableName) throws SQLException {
+        runSqlViaPG("drop table " + tableName);
+        LOG.info().$("dropped table: ").utf8(tableName).$();
+    }
+
+    private static void insertInto(String tableName) throws SQLException {
+        runSqlViaPG("insert into " + tableName + " values('2016-01-01T00:00:00.000Z', 1234)");
+        LOG.info().$("inserted 1 row into table: ").utf8(tableName).$();
+    }
+
+    private static void runSqlViaPG(String sql) throws SQLException {
+        try (
+                final Connection connection = DriverManager.getConnection(PG_CONNECTION_URI, PG_CONNECTION_PROPERTIES);
+                final PreparedStatement stmt = connection.prepareStatement(sql)
+        ) {
+            stmt.execute();
+        }
+    }
+
+    private static void setType(String tableName, String walMode) throws SQLException {
+        runSqlViaPG("alter table " + tableName + " set type " + walMode);
+        LOG.info().$("scheduled table type conversion for table ").utf8(tableName).$(" to ").utf8(walMode).$();
+    }
+
+    private void validateShutdown() throws SQLException {
+        try {
+            insertInto(tableName);
+            fail("Expected exception has not been thrown");
+        } catch (PSQLException psqlException) {
+            assertTrue(psqlException.getMessage().startsWith("Connection to 127.0.0.1:8822 refused."));
+        }
+    }
+}
```

Comment:
could you add a test with this sceneario:

```sql
create table if not exists trade_test (
    sym symbol index capacity 128,
    px long,
    qty int,
    leverage float,
    ts timestamp
) timestamp(ts) partition by day WAL;

tables();
wal_tables();

alter table trade_test set type bypass wal;

tables();
wal_tables();
```

I run this in the web console and if the table is create wal upfront, then it cannot be converted into a non wal, at least `tables()`, `wal_tables()` seem to report otherwise.

Reply:
did this yesterday many times, worked... let me try it.

Reply:
just realised what is going on here...
you probably did not do a restart.

```
alter table trade_test set type bypass wal;
```
this schedules the conversion but does not convert the table.
actual conversion happens only on restart.
maybe we can make `tables()` to check for the presence of the _conversion file and display something to indicate that there is a pending conversion.

Reply:
A restart of the whole server until the next conversion?. Maybe the reader and writer pools could lock the table, awaiting for current locks to be returned, and then perform the conversion?. Otherwise, consider this a nit :)

Reply:
that was the original plan what you just described above, to perform the conversion under full lock.
it was ditched last week.
main concern was how to notify ILP to destroy WalWriters and get TableWriter instead.
there would have been a way but quite convoluted so the restart idea won.
it is assumed that users will not perform these conversions often.